presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -134,8 +133,9 @@ public SqlQueryScheduler(QueryStateMachine queryStateMachine,
         // Only fetch a distribution once per query to assure all stages see the same machine assignments
         Map<PartitioningHandle, NodePartitionMap> partitioningCache = new HashMap<>();
 
+        OutputBufferId rootBufferId = Iterables.getOnlyElement(rootOutputBuffers.getBuffers().keySet());
```

Comment:
Is there always one even when we don't have an output stage?

Reply:
Yes